bot/common/tasks/tasks.py

```diff
@@ -1,25 +1,56 @@
-import threading
 import logging
 import asyncio
+import discord
+import pytz
 
 from abc import ABC, abstractmethod
 from datetime import timedelta, datetime, time
+from discord.ext import tasks, commands
 
 from bot.common.tasks.weekly_contributions import send_weekly_contribution_reports
-from bot.common.cache import Cache, RedisCache
+from bot.common.cache import Cache
+from bot.constants import BotTasks as TaskConstants
 
 logger = logging.getLogger(__name__)
 
-TASK_LOOP_INTERVAL_MINUTES = 10
-REPORT_LAST_SENT_DATETIME_CACHE_KEY = "contribution_report_last_sent"
+# TODO: tz?
 DATETIME_CACHE_FMT = "%m/%d/%Y, %H:%M:%S"
 
 
 class Cadence(ABC):
-    # Returns the number of seconds until the task can be run. Negative
-    # values reflect that the task is due to fire.
+    """
+    Abstract base class for cadences. Objects that implement this ABC
+    are intended to help with repetive tasks that fire on a specific
+    schedule.
+    """
+
+    """
+    Returns the number of seconds until the task can be run. Negative
+    values reflect that the task is due to fire. When negative values
+    are returned are at the discretion of the implementing class.
+    """
+
     @abstractmethod
-    async def get_timedelta_until_run(self, cache: Cache, cache_key: str) -> timedelta:
+    def get_timedelta_until_run(self, from_dt: datetime) -> timedelta:
+        pass
+
+    """
+    Returns the timedelta until the next scheduled occurrence after
+    from_dt. Return value's timedelta is expected to be gte zero (i.e.)
+    timedelta.total_seconds() >= 0.
+    """
```

Comment:
This docstring should go in the function. We are using the style guide below.

https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings

Reply:
brainfart; fixed it up